src/tools/pub/automated-publishing.md

```diff
@@ -0,0 +1,719 @@
+---
+title: Automate publishing packages to pub.dev
+description: Publish Dart packages to pub.dev directly from GitHub Actions.
+---
+
+You can automate publishing package updates to pub.dev using
+GitHub Actions.
+
+{{site.alert.note}}
+  Today, you can only automate publishing of existing packages.
+  To create a new package, publish the first version using
+  `dart pub publish`.
+{{site.alert.end}}
+
+This automation requires authenticating to `pub.dev` using either
+[GitHub-signed OpenID Connect tokens][1] or a [GCP Service Account][2].
+
+With these methods of automated publishing, you don't need to store
+long-lived secrets. The `dart pub` client authenticates using temporary
+tokens that either GitHub Actions or Google Cloud IAM have signed.
+
+
+## Enable publishing packages using GitHub Actions
+
+To publish from [GitHub Actions](https://github.com/features/actions),
+you must:
+
+1. Enable automated publishing for the package on `pub.dev`.
+   You need to provide the following information in `pub.dev`:
+   * Your package's GitHub _repository_ from which you publish the
+     package.
+   * A _`tag-pattern`_ that git tags must match to permit publishing.
+1. Create a GitHub Action workflow for publishing in GitHub.
+1. Create a `pubspec.yaml` file in your repository.
+   This file requires a `version` key with a value that matches the
+   `tag-pattern`.
+1. Create a git tag for your latest commit.
+   This tag also must match the `tag-pattern`.
+
+{{site.alert.note}}
+  GitHub Actions can publish to `pub.dev` only when a push of a git tag
+  to GitHub triggers the Action _workflow_.
+  If a GitHub Actions triggers publishing without a tag, `pub.dev`
```

Comment:
Or we revert back to my text:
>   Pub.dev rejects publishing from GitHub Actions triggered without a tag.

Since new text is grammartically flawed I'm going back.